Subject: Linux-RT 2.6.25-rt
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From efault@gmx.de Mon Jan 14 23:35:16 2008
Date: Mon, 14 Jan 2008 09:27:40 +0100
From: Mike Galbraith <efault@gmx.de>
To: Steven Rostedt <rostedt@goodmis.org>
Cc: Mariusz Kozlowski <m.kozlowski@tuxland.pl>,
     LKML <linux-kernel@vger.kernel.org>, RT <linux-rt-users@vger.kernel.org>,
     Ingo Molnar <mingo@elte.hu>, Thomas Gleixner <tglx@linutronix.de>
Subject: Re: 2.6.24-rc7-rt1

    [ The following text is in the "utf-8" character set. ]
    [ Your display is set for the "iso-8859-1" character set.  ]
    [ Some characters may be displayed incorrectly. ]


On Sun, 2008-01-13 at 15:54 -0500, Steven Rostedt wrote:

> OK, -rt2 will take a bit more beating from me before I release it, so it
> might take some time to get it out (expect it out on Monday).

Ah, that reminds me (tests, yup) I still need the patchlet below to
resume from ram without black screen of death.  No idea why my P4 box
seems to be the only box in the rt galaxy affected.  (haven't poked at
it since the holidays)

---
 kernel/sched_rt.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6.25.2-rt1/kernel/sched_rt.c
===================================================================
--- linux-2.6.25.2-rt1.orig/kernel/sched_rt.c	2008-05-08 16:19:11.000000000 -0400
+++ linux-2.6.25.2-rt1/kernel/sched_rt.c	2008-05-08 16:19:39.000000000 -0400
@@ -33,6 +33,9 @@ static inline void rt_clear_overload(str
 
 static void update_rt_migration(struct rq *rq)
 {
+	if (unlikely(num_online_cpus() == 1))
+		return;
+
 	if (rq->rt.rt_nr_migratory && (rq->rt.rt_nr_running > 1)) {
 		if (!rq->rt.overloaded) {
 			rt_set_overload(rq);
@@ -317,6 +320,7 @@ void dec_rt_tasks(struct sched_rt_entity
 #ifdef CONFIG_SMP
 	if (rt_se->nr_cpus_allowed > 1) {
 		struct rq *rq = rq_of_rt_rq(rt_rq);
+		BUG_ON(!rq->rt.rt_nr_migratory);
 		rq->rt.rt_nr_migratory--;
 	}
 
